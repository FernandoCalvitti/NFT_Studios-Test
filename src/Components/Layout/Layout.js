import React from 'react'
import Header from '../Header/Header'
import ItemList from '../ItemList/ItemList'

const Layout = () => {
    return (
        <div >
            <Header />
            <ItemList />
        </div>
    )
}

export default Layout
