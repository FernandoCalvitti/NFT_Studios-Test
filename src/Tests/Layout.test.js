import React from 'react'
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'
import Layout from '../Components/Layout/Layout'


describe('Layout tests', () => {

    test('Layout matchs snapshots and contains <Header /> and <ItemList />', () => {

        const layout = render(
            <Layout />
        )

        expect(layout).toMatchSnapshot();
    })

})