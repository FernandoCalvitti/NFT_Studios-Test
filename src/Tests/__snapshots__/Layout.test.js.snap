// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout tests Layout matchs snapshots and contains <Header /> and <ItemList /> 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="btn-secondary p-4"
          data-testid="header"
        >
          <h1
            data-testid="title"
          >
            NFT STUDIOS CHALLENGE
          </h1>
          <strong
            data-testid="date"
          >
            2021
          </strong>
        </div>
        <div
          class="container m-4"
        >
          <div
            class="container position-relative top-100 start-50 translate-middle"
            data-testid="spinner"
          >
            <div
              class="spinner-grow p-4 text-primary"
            />
            <div
              class="spinner-grow p-4 text-success"
            />
            <div
              class="spinner-grow p-4 text-danger"
            />
            <div
              class="spinner-grow p-4 text-primary"
            />
            <div
              class="spinner-grow p-4 text-success"
            />
            <div
              class="spinner-grow p-4 text-danger"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="btn-secondary p-4"
        data-testid="header"
      >
        <h1
          data-testid="title"
        >
          NFT STUDIOS CHALLENGE
        </h1>
        <strong
          data-testid="date"
        >
          2021
        </strong>
      </div>
      <div
        class="container m-4"
      >
        <div
          class="container position-relative top-100 start-50 translate-middle"
          data-testid="spinner"
        >
          <div
            class="spinner-grow p-4 text-primary"
          />
          <div
            class="spinner-grow p-4 text-success"
          />
          <div
            class="spinner-grow p-4 text-danger"
          />
          <div
            class="spinner-grow p-4 text-primary"
          />
          <div
            class="spinner-grow p-4 text-success"
          />
          <div
            class="spinner-grow p-4 text-danger"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
